src/Microsoft.Data.SqlClient/tests/tools/TDS/TDS.Servers/GenericTdsServer.cs

```diff
@@ -681,6 +689,40 @@ protected virtual TDSMessageCollection OnAuthenticationCompleted(ITDSServerSessi
                 }
             }
 
+            if (session.IsEnhancedRoutingSupportEnabled &&
+                EnableEnhancedRouting != FeatureExtensionEnablementTriState.DoNotAcknowledge)
+            {
+                // Create ack data (1 byte: IsEnabled)
+                byte[] data = new byte[1];
+
+                if (EnableEnhancedRouting == FeatureExtensionEnablementTriState.Enabled)
```

Comment:
Can has ternary again?

Reply:
In fact, can just do
```
byte[] data = EnableEnhancedRouting is FeatureExtensionEnablementTristate.Enabled
    ? [1]
    : [0];
```

(and if we switch from tristate to bool? we could condense even more to)
```
byte[] data = IsEnhancedRoutingEnabled ? [1] : [0]
```